lldb/source/Plugins/SymbolFile/DWARF/DWARFDIE.cpp

```diff
@@ -404,7 +409,7 @@ std::vector<lldb_private::CompilerContext> DWARFDIE::GetDeclContext() const {
     push_ctx(CompilerContextKind::Enum, GetName());
     break;
   case DW_TAG_subprogram:
-    push_ctx(CompilerContextKind::Function, GetPubname());
+    push_ctx(CompilerContextKind::Function, GetName());
```

Comment:
What's the impact of this change? Looks like previously if a `DW_TAG_subprogram` had a linkage name, we'd put that into the `CompilerContext`? Is that not necessary anymore?

Reply:
IIRC the linkage name / pubname each go into one of dwarf declaration/definition (I forget which one goes where), and thus we have this change?

Reply:
I noticed that Pubname returns the linkage name, which is not useful here, and apparently changing this had no effect on the testsuite.